cd/auto-generate/auto-generate-controllers.sh

```diff
@@ -0,0 +1,241 @@
+#!/usr/bin/env bash
+
+set -eo pipefail
+
+USAGE="
+Usage:
+  $(basename "$0")
+
+Environment variables:
+  PR_SOURCE_BRANCH:    Name of the GitHub branch where auto-generated service
+                       controller code is pushed. Defaults to 'ack-bot-autogen'
+  PR_TARGET_BRANCH:    Name of the GitHub branch where the PR should merge the
+                       code. Defaults to 'main'
+  GH_ORG:              Name of the GitHub organisation where GitHub issues will
+                       be created when autogeneration of service controller fails.
+                       Defaults to 'aws-controllers-k8s'
+  GH_ISSUE_REPO:       Name of the GitHub repository where GitHub issues will
+                       be created when autogeneration of service controller fails.
+                       Defaults to 'community'
+  GH_LABEL:            Label to add to issue and pull requests.
+                       Defaults to 'ack-bot-autogen'
+  GITHUB_ACTOR:        Name of the GitHub account creating the issues & PR.
+  GITHUB_DOMAIN:       Domain for GitHub. Defaults to 'github.com'
+  GITHUB_EMAIL_PREFIX: The 7 digit unique id for no-reply email of
+                       '$GITHUB_ACTOR'
+  GITHUB_TOKEN:        Personal Access Token for '$GITHUB_ACTOR'
+"
+
+# Important Directory references based on prowjob configuration.
+THIS_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+AUTO_GEN_DIR=$THIS_DIR
+CD_DIR=$THIS_DIR/..
+TEST_INFRA_DIR=$CD_DIR/..
+WORKSPACE_DIR=$TEST_INFRA_DIR/..
+CODEGEN_DIR=$WORKSPACE_DIR/code-generator
+
+DEFAULT_PR_SOURCE_BRANCH="ack-bot-autogen"
```

Comment:
`dependabot` uses a branch that starts with `dependabot/`, which is a way of grouping branches. Example: https://github.com/dependabot/dependabot-core/tree/dependabot/composer/composer/helpers/v1/phpstan/phpstan-0.12.98

Would be cool to do something like `ack-bot/runtime`?